### ISSUE TITLE
Auto-detect Analog/Digital

### ISSUE THREAD

**Issue by Catarax** [KrossX/Pokopom#6]
LilyPad recently got updated to auto detect if analog signal is not supported to switch to digital.
Would be nice to include it on Pokopom.

**Comment by KrossX** [comment]
You mean this commit? https://github.com/PCSX2/pcsx2/commit/d736dc1a0a2bdefb15ecc2b742ad0d981c02e2cb

I don't see anything that actually switches to digital mode for incompatible games. All it currently does is toggle analog if current mode is digital when autoanalog is on. I think it just helps autoanalog from overriding other non-digital modes.

**Comment by Catarax** [comment]
Yes this commit, ah that was that.

Well if you use analog in Pokopom by default, some games will display an error message like that :
https://github.com/PCSX2/pcsx2/issues/1348

That's what should be fixed, thanks.

**Comment by KrossX** [comment]
There's an option in Pokopom to set the default mode, it's on digital by default. Other than that, DualShock controllers have an analog button do deal with this problem and that's how Pokopom handles it.

**Comment by shinra358** [comment]
I was just about to come on and ask about this because I had pokopom on analog mode and tried playing FFVI and controller didn't work until I pressed the PS button. On the real console, you don't have to do that to get the controller to work. With EPSXE I mean. When Pokopom is on analog, you have to manually switch it to digital on non-analog games. But if it is on digital, it seems that it auto switches to the correct format if playing an analog game.

**Comment by KrossX** [comment]
There should be no analog/digital option to begin with. The controller should start in digital mode that would be it. But you know emulator usage, quirks and stuff.

Anyhow, a DualShock controller has extra commands from the common digital pad that allows programmers to configure the analog mode and even lock it down. Thus, the controller can toggle from digital to analog mode if the developers of the game made it so. Since the DS controller was introduced later in the life of the console, those newer commands were not available and so the user had to use the analog button to toggle the controller back into a compatible digital mode if necessary or toggle the analog mode on if the game does no auto-detect.

So basically, having to switch to digital mode when playing older games while having set Pokopom to analog as default, is a side effect of having that option.

**Comment by shinra358** [comment]
"There should be no analog/digital option to begin with."
I don't understand. Those options are right there in the pokopom options. What do you mean?

**Comment by KrossX** [comment]
It's a commodity option that causes this sort of issues.
